wp-includes/class-wp-web-app-manifest.php

```diff
@@ -677,4 +707,37 @@ function updateShortNameField() {
 		</script>
 		<?php
 	}
+
+	/**
+	 * Load 'wp/site-logo'-block filter
+	 *
+	 * Handle the 'site_icon_maskable' setting in a full-site-editing context,
+	 * by enqueing a filter to the 'wp:site-logo' block within the site-editor.
+	 *
+	 * @since  0.8.0-alpha
+	 * @see    https://developer.wordpress.org/reference/hooks/enqueue_block_editor_assets/
+	 *
+	 * @return void
+	 */
+	public function enqueue_site_icon_maskable_block_editor_assets() {
+		$dir  = '/wp-includes/js/dist';
+		$path = PWA_PLUGIN_DIR . $dir;
+
+		// Stop, if needed file was not built successfully.
+		$script_asset_path = "$path/site-icon-maskable.asset.php";
+		if ( ! file_exists( $script_asset_path ) ) {
+			return;
+		}
```

Comment:
You are right, that is a redundant check, but leaving it out will result in fatal error with the following `require` statement on the next line. 

Additionally, because this is called inside the block-editor, where the normal `admin_notice`s will not be visible, this should prevent errors while trying to enque a non-existent file. To help the user mentioning, that the plugin was not built properly, we could introduce one of the newer [`createWarningNotice`](https://developer.wordpress.org/block-editor/reference-guides/data/data-core-notices/#createwarningnotice) for the block-editor.

Reply:
> You are right, that is a redundant check, but leaving it out will result in fatal error with the following `require` statement on the next line. Or am I wrong?

If you see the change you made in `pwa.php`:

https://github.com/GoogleChromeLabs/pwa-wp/blob/81d28600276b84d91581cd0969fb9124bf99a978/pwa.php#L100-L106

Note that in the bootstrap, it does a `return` to prevent the rest of the PWA functionality from being loaded. So if the file doesn't exist, then this conditional won't ever be reached.